### PR TITLE
docs: Update performance tips

### PR DESCRIPTION
**Pull request by jeromelaban**

### CODE REVIEW

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.0

File: doc/articles/Uno-UI-Performance.md [hunk 1]
```diff
@@ -4,22 +4,24 @@ uid: Uno.Development.Performance
...
+- Use `x:Load={x:Bind MyVisibility}` where appropriate as toggling from `true` to `false` effectively removes a part of the visual tree from memory. Note that setting back to true re-creates the visual tree.
+- `ListView` and `GridView`
+	- Don't use template selectors inside the ItemTemplate, prefer using the `ItemTemplateSelector` on `ListView`/`GridView`.
+	- The default [`ListViewItem` and `GridViewItem` styles](https://github.com/unoplatform/uno/blob/74b7d5d0e953fcdd94223f32f51665af7ce15c60/src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml#L951) are very feature rich, yet that makes them quite slow. For instance, if you know that you're not likely to use selection features for a specific ListView, create a simpler ListViewItem style that some visual states, or the elements that are only used for selection.
```
Comment (agneszitte) [inline doc/articles/Uno-UI-Performance.md:19]:
```suggestion
	- The default [`ListViewItem` and `GridViewItem` styles](https://github.com/unoplatform/uno/blob/74b7d5d0e953fcdd94223f32f51665af7ce15c60/src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml#L951) are very feature-rich, yet that makes them quite slow. For instance, if you know that you're not likely to use selection features for a specific ListView, create a simpler ListViewItem style that some visual states, or the elements that are only used for selection.
```

File: doc/articles/Uno-UI-Performance.md [hunk 2]
```diff
@@ -85,21 +87,38 @@ Here's what to look for:
...
 	- Enabling the [IL Linker](features/using-il-linker-webassembly.md)
 	- Enabling [XAML Resources Trimming](features/resources-trimming.md)
 
+## Android specifics
+- Adjust the [GC configuration](https://learn.microsoft.com/en-us/xamarin/android/internals/garbage-collection#configuration) by modifying the `environment.conf` file with parameters matching your application
+- [Enable Startup Tracing](https://devblogs.microsoft.com/dotnet/performance-improvements-in-dotnet-maui/#record-a-custom-aot-profile) by running the following:
+	```bash
+	dotnet add package Mono.AotProfiler.Android
+	dotnet build -t:BuildAndStartAotProfiling
+	# Wait until app launches, then navigate around the most common screens
```
Comment (agneszitte) [inline doc/articles/Uno-UI-Performance.md:112]:
```suggestion
	# Wait until the app launches, then navigate around the most common screens
```

File: doc/articles/guides/profiling-applications.md [hunk 3]
```diff
@@ -4,18 +4,105 @@ uid: Uno.Tutorials.ProfilingApplications
...
+```bash
+alias mlaunch=/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/bin/mlaunch
+```
+
+### Profiling on a iOS Simulator
+
+1. The first step is to launch the tool that provides a connection between the app and the .NET tracing tools:
+
+    ```bash
+    $ dotnet-dsrouter client-server -ipcc ~/my-sim-port -tcps 127.0.0.1:9000
```
Comment (agneszitte) [inline doc/articles/guides/profiling-applications.md:33]:
```suggestion
## Profiling .NET iOS applications

> [!NOTE]
> This documentation is based on [.NET iOS profiling documentation](https://github.com/xamarin/xamarin-macios/wiki/Profiling).

Profiling iOS apps needs to be done on a Mac machine/

First, create an alias to mlaunch:
```bash
alias mlaunch=/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/bin/mlaunch
```

### Profiling on an iOS Simulator

1. The first step is to launch the tool that provides a connection between the app and the .NET tracing tools:

    ```bash
    $ dotnet-dsrouter client-server -ipcc ~/my-sim-port -tcps 127.0.0.1:9000
```

File: doc/articles/guides/profiling-applications.md [hunk 4]
```diff
@@ -4,18 +4,105 @@ uid: Uno.Tutorials.ProfilingApplications
...
+$ mlaunch --installdev bin/Debug/net*/*/*.app --devname ... 
+$ mlaunch --launchdev bin/Debug/net*/*/*.app --devname ... --wait-for-exit --argument --connection-mode --argument none '--setenv:DOTNET_DiagnosticPorts=127.0.0.1:9001,suspend,listen'
+```
+
+At this point it's necessary to wait until the following line shows up in the terminal:
+
+```
+The runtime has been configured to pause during startup and is awaiting a Diagnostics IPC ResumeStartup command from a Diagnostic Port
+```
+Once that's printed, go ahead and start profiling:
```
Comment (agneszitte) [inline doc/articles/guides/profiling-applications.md:83]:
```suggestion
Install & launch the app and make it suspended upon launch:

```bash
$ mlaunch --installdev bin/Debug/net*/*/*.app --devname ... 
$ mlaunch --launchdev bin/Debug/net*/*/*.app --devname ... --wait-for-exit --argument --connection-mode --argument none '--setenv:DOTNET_DiagnosticPorts=127.0.0.1:9001,suspend,listen'
```

At this point, it's necessary to wait until the following line shows up in the terminal:

```
The runtime has been configured to pause during startup and is awaiting a Diagnostics IPC ResumeStartup command from a Diagnostic Port
```
Once that's printed, go ahead and start profiling:
```

File: doc/articles/guides/profiling-applications.md [hunk 5]
```diff
@@ -4,18 +4,105 @@ uid: Uno.Tutorials.ProfilingApplications
...
+> This documentation is based on [.NET iOS profiling documentation](https://github.com/xamarin/xamarin-macios/wiki/Profiling).
+
+Profiling iOS apps needs to be done on a mac machine/
+
+First, create an alias to mlaunch:
+```bash
+alias mlaunch=/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/bin/mlaunch
+```
+
+### Profiling on a iOS Simulator
```
Comment (agneszitte) [inline doc/articles/guides/profiling-applications.md:28]:
```suggestion
### Profiling on an iOS Simulator
```

File: doc/articles/guides/profiling-applications.md [hunk 6]
```diff
@@ -4,18 +4,105 @@ uid: Uno.Tutorials.ProfilingApplications
...
+```
+
+Install & launch the app and make it suspend upon launch:
+
+```bash
+$ mlaunch --installdev bin/Debug/net*/*/*.app --devname ... 
+$ mlaunch --launchdev bin/Debug/net*/*/*.app --devname ... --wait-for-exit --argument --connection-mode --argument none '--setenv:DOTNET_DiagnosticPorts=127.0.0.1:9001,suspend,listen'
+```
+
+At this point it's necessary to wait until the following line shows up in the terminal:
```
Comment (agneszitte) [inline doc/articles/guides/profiling-applications.md:78]:
```suggestion
At this point, it's necessary to wait until the following line shows up in the terminal:
```

File: doc/articles/guides/profiling-applications.md [hunk 7]
```diff
@@ -4,18 +4,105 @@ uid: Uno.Tutorials.ProfilingApplications
...
+
+### Profiling on a physical iOS device
+
+Launch the tool that bridges the app and the .NET tracing tools:
+
+```bash
+$ dotnet-dsrouter server-client -ipcs ~/my-dev-port -tcpc 127.0.0.1:9001 --forward-port iOS
+```
+
+Install & launch the app and make it suspend upon launch:
```
Comment (agneszitte) [inline doc/articles/guides/profiling-applications.md:71]:
```suggestion
Install & launch the app and make it suspended upon launch:
```

File: doc/articles/guides/profiling-applications.md [hunk 8]
```diff
@@ -4,18 +4,105 @@ uid: Uno.Tutorials.ProfilingApplications
...
 Run the following commands
 - `dotnet tool update -g dotnet-dsrouter --add-source=https://aka.ms/dotnet-tools/index.json`
 - `dotnet tool update -g dotnet-trace --add-source=https://aka.ms/dotnet-tools/index.json`
 
+## Profiling .NET iOS applications
+
+> [!NOTE]
+> This documentation is based on [.NET iOS profiling documentation](https://github.com/xamarin/xamarin-macios/wiki/Profiling).
+
+Profiling iOS apps needs to be done on a mac machine/
```
Comment (ajpinedam) [inline doc/articles/guides/profiling-applications.md:21]:
```suggestion
Profiling iOS apps needs to be done on a mac machine.
```

**Review by ajpinedam — APPROVED** [review]

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**